libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsStorage.java

```diff
@@ -70,25 +71,20 @@ public BlobsStorage(String dataConnectionString, String containerName) {
             throw new IllegalArgumentException("The 'keys' parameter is required.");
         }
 
-        initialize();
-
         Map<String, Object> items = new HashMap<>();
 
         for (String key : keys) {
             try {
                 String blobName = GetBlobName(key);
                 BlobClient blobClient = containerClient.getBlobClient(blobName);
-                items.put(key, innerReadBlob(blobClient).get());
+                items.put(key, innerReadBlob(blobClient).join());
```

Comment:
are we using blocking calls after all? Honestly asking

Reply:
Not really, the use of `join()` is wrong here